portfolio/src/main/webapp/script.js

```diff
@@ -12,12 +12,51 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-window.onload = () => { 
+function loadPage() {
+  renderRandomGreeting();
+  renderAuthentication();
+}
+
+function renderAuthentication(){
+  fetch('/authenticate').then((response) => response.json()).then((response) => {
+    debugLog(response);
+    const navbarList = document.getElementById('navbar-list');
+    if(response.isUserLoggedIn){
+      navbarList.appendChild(createAuthenticatedStatusElement(response.logoutUrl, "Logout"));
+    } else {
+      navbarList.appendChild(createAuthenticatedStatusElement(response.loginUrl, "Login"));
+    }
+  });
+}
+
+function createAuthenticatedStatusElement(href, text) {
+  const listElement = document.createElement("li"); 
+  listElement.classList.add("nav-item");
+  listElement.classList.add("active");
+
+  const link = document.createElement("a"); 
+  link.classList.add("nav-link");
+  link.setAttribute("href", href);
+  link.innerHTML = text;
+
+  listElement.appendChild(link);
+  return listElement;
+}
```

Comment:
Ideally, you would actually have these elements specified in HTML with `display: none;` set in their style and then show them conditionally by setting the style to `display: inline;`. This allows you to do all of the styling in HTML rather than this somewhat obtuse JavaScript. However, the current approach is fine to leave as-is here.